public/app/controllers/resources_controller.rb

```diff
@@ -299,6 +301,14 @@ def has_digital_objects?
     archivesspace.search(qry, 1).records.any?
   end
 
+  def get_digital_object_count
```

Comment:
There's some code duplication here ... then I looked at the controller in general again and, yeah, this ain't much compared with that and probably not the time to be dealing with it 😅 

I haven't double checked at all but I am surprised (and it seems counter-intuitive) that pagination has any bearing on `total_hits`. It isn't equivalent to `numFound`? Ok, couldn't resist taking a quick and it appears to be: https://github.com/archivesspace/archivesspace/blob/master/backend/app/model/solr.rb#L419 -- so weird but ignorable (I think).

Reply:
We need a test for this, but I'm unsure of what the difference is between the earlier naive approach (json.instances.length) vs. the Solr query. Is it that instances.length only tells you the number of instances for that particular resource, but the solr query finds all of the linked instances in the resource's archival objects too?

Reply:
> We need a test for this, but I'm unsure of what the full difference is between the earlier naive approach (json.instances.length) vs. the Solr query. Is it that instances.length only tells you the number of instances for that particular resource, but the solr query finds all of the linked instances in the resource's archival objects too?

I believe that is more or less the case. The Solr query finds all instances that have the relevant collection uri in them. I can tell you in my quick test with the Rush papers that json.instances.length was returning 0, while the Solr search returns the correct 147.

Reply:
> There's some code duplication here ... then I looked at the controller in general again and, yeah, this ain't much compared with that and probably not the time to be dealing with it 😅
> 
> I haven't double checked at all but I am surprised (and it seems counter-intuitive) that pagination has any bearing on `total_hits`. It isn't equivalent to `numFound`? Ok, couldn't resist taking a quick and it appears to be: https://github.com/archivesspace/archivesspace/blob/master/backend/app/model/solr.rb#L419 -- so weird but ignorable (I think).

You're correct. I did a quick test without the `:page_size` specified and I got the full number. The details are already escaping me, but at some point I must have been using a different field or taking the length of something, and I kept getting back 10 until I added the page size. I must have noticed `total_hits` afterward and thought "oh this looks convenient" and switched it over without realizing it freed me from the pagination.